### PR TITLE
docs(storybook): add controls to combo-box and codesnippet

### LINKED ISSUE #11471: Update CodeSnippet.stories.js to include a playground story and interactive controls
**Checklist**

- [ ] Move `.stories.js` file out of `next` folder, or create `.stories.js` file and copy contents of `*-story.js` file over.
- [ ] Remove deprecated `knobs` package and usage.
- [ ] Ensure there is a playground story
- [ ] Add `argTypes` to playground story
- [ ] `bind()` it as a template as described in the storybook docs
- [ ] Pass `...args` to playground story, spread them on the applicable component(s)
- [ ] Remove deprecated props from being visible in controls of storybook

**Note: if the story has sub-components, document and share with @sstrubberg**

**Links & Resources**

- [DataTable example](https://github.com/carbon-design-system/carbon/pull/10999)

### LINKED ISSUE #11472: Update ComboBox.stories.js to include a playground story and interactive controls
**Checklist**

- [ ] Move `.stories.js` file out of `next` folder, or create `.stories.js` file and copy contents of `*-story.js` file over.
- [ ] Remove deprecated `knobs` package and usage.
- [ ] Ensure there is a playground story
- [ ] Add `argTypes` to playground story
- [ ] `bind()` it as a template as described in the storybook docs
- [ ] Pass `...args` to playground story, spread them on the applicable component(s)
- [ ] Remove deprecated props from being visible in controls of storybook

**Note: if the story has sub-components, document and share with @sstrubberg**

**Links & Resources**

- [DataTable example](https://github.com/carbon-design-system/carbon/pull/10999)

### PR DESCRIPTION
**Pull request by aledavila**
Closes #11472 
Closes #11471 

Adds controls and playground story to ComboBox and Codesnippet

#### Changelog

**New**

- playground stories

**Changed**

- adds certain controls to the playground stories only

#### Testing / Reviewing

- make sure stories list all props and playground only has controls

### CODE REVIEW

File: packages/react/src/components/ComboBox/ComboBox.stories.js [hunk 1]
```diff
@@ -106,9 +103,126 @@ export const WithLayer = (args) => (
...
+      helperText="Combobox helper text"
+      {...args}
+    />
+  </div>
+);
+
+Playground.argTypes = {
+  ariaLabel: {
+    table: {
+      disable: true,
```
Comment (tay1orjones) [inline packages/react/src/components/ComboBox/ComboBox.stories.js:135]: What's the reasoning for excluding this one?

File: packages/react/src/components/ComboBox/ComboBox.stories.js [hunk 2]
```diff
@@ -106,9 +103,126 @@ export const WithLayer = (args) => (
...
+    },
+  },
+  initialSelectedItem: {
+    table: {
+      disable: true,
+    },
+  },
+  invalidText: {
+    table: {
+      disable: true,
```
Comment (tay1orjones) [inline packages/react/src/components/ComboBox/ComboBox.stories.js]: What's the reason for excluding this one?

File: packages/react/src/components/ComboBox/ComboBox.stories.js [hunk 3]
```diff
@@ -106,9 +103,126 @@ export const WithLayer = (args) => (
...
+    },
+  },
+  light: {
+    table: {
+      disable: true,
+    },
+  },
+  onChange: {
+    table: {
+      disable: true,
```
Comment (tay1orjones) [inline packages/react/src/components/ComboBox/ComboBox.stories.js]: For all the callback props, I think if you define them in the component level argTypes up top they'll get logged out to the `Actions` panel. https://storybook.js.org/docs/react/essentials/actions

```js
argTypes: { onChange: { action: 'clicked' } },
```

File: packages/react/src/components/ComboBox/ComboBox.stories.js [hunk 4]
```diff
@@ -106,9 +103,126 @@ export const WithLayer = (args) => (
...
+    },
+  },
+  titleText: {
+    table: {
+      disable: true,
+    },
+  },
+  type: {
+    table: {
+      disable: true,
```
Comment (tay1orjones) [inline packages/react/src/components/ComboBox/ComboBox.stories.js:224]: What's the reason for excluding this one?

File: packages/react/src/components/ComboBox/ComboBox.stories.js [hunk 5]
```diff
@@ -106,9 +103,126 @@ export const WithLayer = (args) => (
...
+    },
+  },
+  warnText: {
+    table: {
+      disable: true,
```
Comment (tay1orjones) [inline packages/react/src/components/ComboBox/ComboBox.stories.js]: What's the reason for excluding this one?

File: packages/react/src/components/CodeSnippet/CodeSnippet.stories.js [hunk 6]
```diff
@@ -220,3 +220,32 @@ export const Skeleton = () => (
...
+    },
+  },
+  children: {
+    table: {
+      disable: true,
+    },
+  },
+  onClick: {
+    table: {
+      disable: true,
```
Comment (tay1orjones) [inline packages/react/src/components/CodeSnippet/CodeSnippet.stories.js:243]: Same thing here, we can define the callback handlers in the component level argTypes and to get them logged out to the `Actions` panel

**Review by tay1orjones — APPROVED** [review]
Overall LGTM - just a couple small potential improvements, and a few questions why some are being excluded from the table.

Very possible I'm just out of touch with the decision/direction of which props we want to include vs not include 😅

**Review by abbeyhrt — APPROVED** [review]

File: packages/react/src/components/ComboBox/ComboBox.stories.js (hunk 4 above)
Comment (aledavila) [reply packages/react/src/components/ComboBox/ComboBox.stories.js:224]: I've updated the others this one specifically was that I couldnt get it to work even using a select control on it with both variants

Comment (tay1orjones) [reply packages/react/src/components/ComboBox/ComboBox.stories.js:224]: Sounds good 👍

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**